components/tidb_query_vec_expr/src/rpn_expr/types/expr_builder.rs

```diff
@@ -153,8 +151,7 @@ impl RpnExpressionBuilder {
 
     /// Pushes a `Constant` node. The field type will be auto inferred by choosing an arbitrary
     /// field type that matches the field type of the given value.
-    #[cfg(test)]
-    pub fn push_constant(mut self, value: impl Into<ScalarValue>) -> Self {
+    pub fn push_constant_for_test(mut self, value: impl Into<ScalarValue>) -> Self {
```

Comment:
Do we have ways to access these members marked as `#[cfg(test)]` in other crates? @kennytm @nrc

Reply:
you could use `cfg(feature = ...)` and enable the feature only in dev-dependency.

but why these methods need to be hidden behind `cfg(test)`?

Reply:
Thanks! For example, In real scenario Coprocessor don't need to build an expression. Instead it only receives an expression from TiDB. Thus utilities to build an expression are marked as test only to avoid abuse and confusion, and allow possibility of not being high performance as a test utility.

Reply:
I tried to use `cfg(feature= 'testing')`, but I did not work it out. Code still not compile.

Reply:
We have a lot of code in TiKV already which is only there for tests but can't be cfg'ed out because it is in a different crate. I wouldn't worry about adding a feature for this.